R/tweak-initial-estimates.R

```diff
@@ -0,0 +1,192 @@
+
+#' Tweak the initial parameter estimates
+#'
+#' @param .mod model object to update.
+#' @param .p Percent to tweak the initial parameter estimates by. Represented as
+#'   a decimal.
+#' @param tweak type of estimates to tweak in the model. Only `"theta"` is
+#'   currently supported, though omega and sigma will be options at a later date.
+#' @param digits Number of significant digits to round estimates to.
+#'
+#' @details
+#'
+#' In the following cases, the initial estimate will *not* be updated:
+#'  - **Individual** `FIXED` `THETA` parameters
+#'     - e.g., `$THETA 1.2 FIX 1.5 0.2` --> would only skip the first value
+#'  - `THETA` parameters with no initial estimate
+#'     - e.g., `$THETA (0,,1)`
+#'
+#'  For bounded `THETA` estimates:
+#'   - If an initial `THETA` has bounds **and** an initial estimate
+#'  (e.g., `(0, 0.5, 1)`, `(0,1)`), the bounds will be respected when sampling
+#'  a percent to tweak by. If the tweaked value would fall below the lower bound,
+#'  the initial estimate will be set to 90% of the way between the initial value
+#'  and lower bound. If *after rounding*, the value would still not be within the
+#'  bounds, the difference between the initial value and the bound will be
+#'  iteratively reduced  until it is. The same is true for upper bounds.
+#'     - e.g., `(0, 0.5, 1)` --> tweak initially, falls outside bound (`(0, 1.2, 1)`)
+#'     --> set to value within bounds (first iteration: `(1-0.5)*0.9)`:  (`(0, 0.45, 1)`)
```

Comment:
I'm pretty sure the end of this should say `(0, 0.95, 1)` not `(0, 0.45, 1)`. I was worried that we had implemented it wrong, but I ran through the code a bit and I think the code is right and these docs are just wrong. Let's talk through this together next week though.

Reply:
Confirmed you were right, updated docs in [490f4ca](https://github.com/metrumresearchgroup/bbr/pull/646/commits/490f4ca2e472d796240576143c8698e1d3de343b)